mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactTableView.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { getSrc } from './ShowArtifactPage';
+import { getArtifactContent } from '../../../common/utils/ArtifactUtils';
+import { Table } from '@databricks/design-system';
+import Papa from 'papaparse';
+
+class ShowArtifactTableView extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchArtifacts = this.fetchArtifacts.bind(this);
+  }
+
+  static MAX_ROW_LENGTH = 500;
+
+  static propTypes = {
+    runUuid: PropTypes.string.isRequired,
+    path: PropTypes.string.isRequired,
+    getArtifact: PropTypes.func,
+  };
+
+  static defaultProps = {
+    getArtifact: getArtifactContent,
+  };
+
+  state = {
+    loading: true,
+    error: undefined,
+    data: undefined,
+    headers: undefined,
+    originalRowLength: 0,
+    text: undefined,
+  };
+
+  componentDidMount() {
+    this.fetchArtifacts();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.path !== prevProps.path || this.props.runUuid !== prevProps.runUuid) {
+      this.setState({ data: undefined, headers: undefined, originalRowLength: 0, loading: true });
+      this.fetchArtifacts();
+    }
+  }
+
+  render() {
+    if (this.state.loading) {
+      return <div className='artifact-text-view-loading'>Loading...</div>;
+    }
+    if (this.state.error) {
+      return (
+        <div className='artifact-text-view-error'>
+          Oops we couldn't load your file because of an error.
+        </div>
+      );
+    }
+
+    if (this.state.data) {
+      // eslint-disable-next-line max-len
+      const rowPreviewMessage = `Previewing the first ${this.state.data.length} rows out of ${this.state.originalRowLength}`;
```

Comment:
Can we use `FormattedMessage` for internationalization?

Example:

https://github.com/mlflow/mlflow/blob/4c0f21025322f4681e5c4cf01d2ad57e0eb93115/mlflow/server/js/src/experiment-tracking/components/MetricView.js#L28-L35